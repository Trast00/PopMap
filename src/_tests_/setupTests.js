import '@testing-library/jest-dom';
import fetchMock from 'fetch-mock';

export const testListCountry = [
  {
    name: 'test country 1',
    population: 1000,
    flags: {
      svg: 'test image url',
    },
    numericCode: 'test1',
  },
  {
    name: 'test country 2',
    population: 2000,
    flags: {
      svg: 'test image url',
    },
    numericCode: 'test2',
  },
];

export const fetchDataMock = () => {
  fetchMock.getOnce('https://restcountries.com/v2/all', {
    headers: { 'content-type': 'application/json' },
    body: testListCountry,
  });
};
