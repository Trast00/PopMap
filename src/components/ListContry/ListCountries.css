.list-country {
  flex-wrap: wrap;
}

.list-country li {
  width: 50%;
}

.list-country li:nth-child(4n),
.list-country li:nth-child(4n+1) {
  background-color: var(--color-primary);
}

.list-country li:nth-child(4n+2),
.list-country li:nth-child(4n+3) {
  background-color: var(--color-primary-dark);
}

.country {
  width: 100%;
  min-height: 200px;
  justify-content: space-evenly;
}

.country img {
  width: 50%;
  opacity: 0.8;
  border: 4px solid white;
  border-radius: 4px;
}

.country .detail-wrapper {
  width: 100%;
  align-items: flex-end;
  text-align: end;
  padding-right: 16px;
}

.country .detail-wrapper h4 {
  color: white;
  font-size: 1.2rem;
  font-weight: 500;
}

.country .detail-wrapper p {
  color: white;
  font-size: 0.9rem;
  font-weight: 400;
}
