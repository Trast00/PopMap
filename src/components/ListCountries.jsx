import React from 'react'

function ListCountries() {
  return (
    <div>ListoConutries</div>
  )
}

export default ListCountries
