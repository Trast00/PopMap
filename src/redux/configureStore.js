import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './country/countryReducer';

export default configureStore({
  reducer: {
    countryReducer,
  },
});
