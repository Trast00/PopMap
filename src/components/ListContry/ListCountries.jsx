import React from 'react'
import Country from './Country'
import "./ListCountries.css"

function ListCountries() {
  return (
    <div className='max-width'>
      <ul className='max-width no-styling row list-country'>
        <Country />
        <Country />
        <Country />
      </ul>
    </div>
  )
}

export default ListCountries
