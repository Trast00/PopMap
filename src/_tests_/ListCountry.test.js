import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { testListCountry } from './setupTests.js';
import { Provider } from 'react-redux';
import configureStore from '../redux/configureStore.js';
import ListCountries from '../components/ListContry/ListCountries.jsx';

describe('test render', () => {
  test('List Country should match snapshoot', () => {
    const tree = renderer.create(
      <React.StrictMode>
        <Provider store={configureStore}>
          <ListCountries country={testListCountry[0]} />
        </Provider>
      </React.StrictMode>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})