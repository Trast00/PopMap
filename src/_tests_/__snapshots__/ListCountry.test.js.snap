// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render List Country should match snapshoot 1`] = `
<div
  className="max-width"
>
  <ul
    className="max-width no-styling row list-country"
  >
    <p
      className="max-width flex-center"
    >
      Empty List
    </p>
  </ul>
</div>
`;
